disttask/loaddata/proto.go

```diff
@@ -29,29 +29,43 @@ const (
 
 // TaskMeta is the task of LoadData.
 type TaskMeta struct {
-	Table     Table
-	Format    Format
-	Dir       string
-	FileInfos []FileInfo
+	JobID       int64
+	Table       Table
+	Format      Format
+	Dir         string
+	FileInfos   []FileInfo
+	Mode        Mode
+	SessionVars SessionVars
+	Stmt        string
 }
 
 // SubtaskMeta is the subtask of LoadData.
 // Dispatcher will split the task into subtasks(FileInfos -> Chunks)
 type SubtaskMeta struct {
-	Table  Table
-	Format Format
-	Dir    string
-	Chunks []Chunk
+	JobID       int64
+	ID          int32
+	Table       Table
+	Format      Format
+	Dir         string
+	Chunks      []Chunk
+	Mode        Mode
+	SessionVars SessionVars
+	Stmt        string
 }
 
 // MinimalTaskMeta is the minimal task of LoadData.
 // Scheduler will split the subtask into minimal tasks(Chunks -> Chunk)
 type MinimalTaskMeta struct {
-	Table  Table
-	Format Format
-	Dir    string
-	Chunk  Chunk
-	Writer backend.EngineWriter
+	JobID       int64
+	Table       Table
+	Format      Format
+	Dir         string
+	Chunk       Chunk
+	Mode        Mode
+	SessionVars SessionVars
+	Stmt        string
+	DataEngine  *backend.OpenedEngine
```

Comment:
I slightly prefer we use another MinimalTask struct to wrap on MinimalTaskMeta, and move non-serializable objects into MinimalTask